functions/config-versioning.bash

```diff
@@ -0,0 +1,310 @@
+  then
+    echo "OK"
+  else
+    return 0
+  fi
+
+  echo -n "$(timestamp) [openHABian] Uninstalling Git Versioning..."
+  if cond_redirect rm -rf /.git; then echo -n ".OK"; else echo ".FAILED (remove .git)"; return 1; fi
+  if cond_redirect rm /.gitignore; then echo -n ".OK"; else echo ".FAILED (remove .gitignore)"; return 1; fi
+  if cond_redirect rm /openhabian_config_versioning; then echo -n ".OK"; else echo ".FAILED (remove install check)"; return 1; fi
+  echo ".Successfull"
+
+  if [[ -n $INTERACTIVE ]]
+  then
+    whiptail --title "Operation Successfull" --msgbox "Local Versioning Removed" 5 80
+  fi
+}
+
+## Enable timer to run commit.
+## This function INTERACTIVE
+## 
+##    versioning_enable_auto_commit()
+##
+versioning_enable_auto_commit() {
+
+  # Can run ./functions/config-versioning.bash commit 
+  # if script has exicute permission for runnin in openhab rule
+
+  true
+
+}
+
+## Disable timer to run commit.
+## This function INTERACTIVE 
+## 
+##    versioning_disable_auto_commit()
+##
+versioning_disable_auto_commit() {
+
+  true
+
+}
+
+## Function to commiting changes and PUSH to github repository for config versioning.
+## This function INTERACTIVE 
+## 
+##    versioning_commit()
+##
+versioning_commit() {
+
+  local commitMessage example exitcode=0
+
+  if [ -d "/.git" ]
+  then 
+    echo "[openHABian] Start Repositiory commit... "
+    example="$(date) Manual Commit"
+    if [[ -n $INTERACTIVE ]];  then  commitMessage="$(whiptail --title "Versioning github commit message" --inputbox "\\nCommit Message" 8 80 "$example" 3>&1 1>&2 2>&3)"
+      exitcode=$?
+    fi
+    if [ $exitcode != 0 ]
+    then
+      return 0
+    else  
+      if [ -z "$commitMessage" ]; then  commitMessage="$(date) Auto Commit"; fi
+      commit=$(git -C "/" cmp "$commitMessage")
+      if [[ -n $INTERACTIVE ]];  then  whiptail --title "Versioning github" --scrolltext --msgbox "$commit" 25 80;  fi
+    fi
+  else
+    echo "[openHABian] No Repositiory Found to commit... "
+    return 0
+  fi
+
+}
+
+## Function is to update/change the github token used in config versioning.
+## This functions only in INTERACTIVE
+## 
+##    versioning_commit()
+##
+versioning_tokenupdate() {
+
+  local newtoken repositiory user newrepourl
+
+  newtoken="$(whiptail --title "Versioning Github" --inputbox "\\nGithub Token?" 8 39 --title "Enter new Token" 3>&1 1>&2 2>&3)"
+  repositiory=$(git -C "/" config versioning.repositiory)
+  user=$(git -C "/" config user.user)
+  newrepourl="https://$user:$newtoken@github.com/$user/$repositiory.git"
+  echo "$newrepourl"
+  echo -n "$(timestamp) [openHABian] Check if new token is valid..."
+  if git ls-remote "$newrepourl" | grep "HEAD" > /dev/null 
+  then
+    git -C "/" remote set-url origin "$newrepourl"
+    git -C "/" config versioning.token "$newtoken"
```

Comment:
will that permanently store the access token so it is not being asked for again ?